core/iwasm/common/wasm_memory.c

```diff
@@ -765,6 +776,17 @@ wasm_enlarge_memory_internal(WASMModuleInstance *module, uint32 inc_page_count)
     bh_assert(total_size_new
               <= GET_MAX_LINEAR_MEMORY_SIZE(memory->is_memory64));
 
+#if WASM_MEM_ALLOC_WITH_USAGE != 0
+    (void)full_size_mmaped;
+    if (!(memory_data_new = realloc_func(Alloc_For_LinearMemory,
+                                         memory_data_old, total_size_new))) {
+        ret = false;
+        goto return_func;
+    }
+    memory->heap_data = memory_data_new + (heap_data_old - memory_data_old);
```

Comment:
The heap data (base address) may be changed, should call `mem_allocator_migrate` for it like below code since the internal pointers (absolute addresses) inside the heap should be updated, or they will point to invalid buffer. If not, allocation by wasm app or module_malloc by host may fail.

Reply:
OK